iis3dwb_reg.c

```diff
@@ -165,9 +161,8 @@ int32_t iis3dwb_xl_full_scale_set(stmdev_ctx_t *ctx,
                                   iis3dwb_fs_xl_t val)
 {
   iis3dwb_ctrl1_xl_t ctrl1_xl;
-  int32_t ret;
 
-  ret = iis3dwb_read_reg(ctx, IIS3DWB_CTRL1_XL, (uint8_t *)&ctrl1_xl, 1);

```

Comment:
not getting the reason for this.
CAn you explain it a bit?

Reply:
Defining a variable when it can be initialized to a meaningful value it's a best practise.
It's the opposite of what is known as ITM (Initialize-Then-Modify) anti-pattern.
The compiler is smart enough to not generate boilerplate code and almost certainly allocates and initializes the variable only once.
But for the (human) developer, the code is more expressive. Anmd, sometimes, the developer sees that the variable can be defined as `const`  (impossible to do if split over two instructions). "const-qualify everything" is another well-kown pattern.
This is the reason why static analyzers give this kind of warnings.